File: library/src/main/java/com/afollestad/aesthetic/NavigationViewMode.java
package com.afollestad.aesthetic;

/** @author Aidan Follestad (afollestad) */
public enum NavigationViewMode {
  SELECTED_PRIMARY(0),
  SELECTED_ACCENT(1);

  private final int value;

  NavigationViewMode(int value) {
    this.value = value;
  }

  public int toInt() {
    return value;
  }

  public static NavigationViewMode fromInt(int value) {
    switch (value) {
      case 0:
        return SELECTED_PRIMARY;
      default:
        return SELECTED_ACCENT;
    }
  }
}

File: library/src/main/java/com/afollestad/aesthetic/NavigationViewMode.kt
package com.afollestad.aesthetic

/** @author Aidan Follestad (afollestad) */
enum class NavigationViewMode(val value: Int) {
  SELECTED_PRIMARY(0),
  SELECTED_ACCENT(1);

  companion object {
    fun fromInt(value: Int): NavigationViewMode {
      return when (value) {
        0 -> SELECTED_PRIMARY
        else -> SELECTED_ACCENT
      }
    }
  }
}
